network-api/networkapi/wagtailpages/pagemodels/customblocks/common/link_blocks.py

```diff
@@ -0,0 +1,43 @@
+"""
+These are utility blocks used for StructBlocks related to MozFest.
+"""
+
+from wagtail.core import blocks
+
+
+class InternalLinkBlockStructValue(blocks.StructValue):
+    @property
+    def link_url(self):
+        return self.get('link').url
+
+
+class ExternalLinkBlockStructValue(blocks.StructValue):
+    @property
+    def link_url(self):
+        return self.get('link')
+
+
+class InternalLinkBlock(blocks.StructBlock):
+    link = blocks.PageChooserBlock(help_text='Page that this should link out to.')
+
+    class Meta:
+        value_class = InternalLinkBlockStructValue
+
+
+class ExternalLinkBlock(blocks.StructBlock):
+    link = blocks.URLBlock(help_text='URL that this should link out to.')
+
+    class Meta:
+        value_class = ExternalLinkBlockStructValue
```

Comment:
Does having this not work?

```python
class InternalLinkBlock(blocks.StructBlock):
    link = blocks.PageChooserBlock(...)

    @property
    def link_url(self):
        return self.get('link')
```

Or is the goal to make `link` completely unavailable outside of these blocks?

Reply:
This won't work for StructBlocks unfortunately: https://docs.wagtail.io/en/latest/advanced_topics/customisation/streamfield_blocks.html#additional-methods-and-properties-on-structblock-values

Adding properties to a StructBlock requires you to create a StructValue